### PR TITLE
[Feature] Support DETR3D in mmdet3d-v1.0.0

### PR DESCRIPTION
**Pull request by ZLTJohn**
## Motivation

Add DETR3D for mmdet3d-v1.0.0

## Modification

Add DETR3D in projects/detr3d

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: projects/detr3d/mmdet3d_plugin/core/bbox/assigners/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from .hungarian_assigner_3d import HungarianAssigner3D
```
Comment (ZwwWayne) [inline projects/detr3d/mmdet3d_plugin/core/bbox/assigners/__init__.py:1]: We can put all files into projects/detr3d/detr3d directory

Comment (ZwwWayne) [reply projects/detr3d/mmdet3d_plugin/core/bbox/assigners/__init__.py:1]: This way, __init__.py can be omitted and the file structure can be simplified.

File: projects/detr3d/mmdet3d_plugin/core/bbox/assigners/hungarian_assigner_3d.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+            bbox_pred.device)
+
+        # 4. assign backgrounds and foregrounds
+        # assign all indices to backgrounds first
+        assigned_gt_inds[:] = 0
+        # assign foregrounds based on matching results
+        assigned_gt_inds[matched_row_inds] = matched_col_inds + 1
+        assigned_labels[matched_row_inds] = gt_labels[matched_col_inds]
+        return AssignResult(
+            num_gts, assigned_gt_inds, None, labels=assigned_labels)
```
Comment (ZwwWayne) [inline projects/detr3d/mmdet3d_plugin/core/bbox/assigners/hungarian_assigner_3d.py:134]: file format issue. The file should ad least pass pre-commit hook

File: projects/detr3d/configs/nuscene/detr3d_res101_gridmask.py [hunk 3]
```diff
@@ -0,0 +1,221 @@
+_base_ = [
+    '../../../../configs/_base_/datasets/nus-3d.py',
+    '../../../../configs/_base_/default_runtime.py'
+]
+
+custom_imports = dict(imports=['projects.detr3d.mmdet3d_plugin'])
```
Comment (JingweiZhang12) [inline projects/detr3d/configs/nuscene/detr3d_res101_gridmask.py:6]: Most of the comments are the same as https://github.com/open-mmlab/mmdetection3d/pull/2170. Please refine this PR like https://github.com/open-mmlab/mmdetection3d/pull/2170.

**Pull request closed without merging**